src/assets/scripts/templates/customers/addresses.js

```diff
@@ -8,31 +8,37 @@
  */
 
 import $ from 'jquery';
+import {CountryProvinceSelector} from '@shopify/theme-addresses';
 
 const $newAddressForm = $('#AddressNewForm');
 
 if ($newAddressForm.length) {
-  // Initialize observers on address selectors, defined in shopify_common.js
-  if (Shopify) {
-    new Shopify.CountryProvinceSelector(
-      'AddressCountryNew',
-      'AddressProvinceNew',
-      {
-        hideElement: 'AddressProvinceContainerNew',
-      },
-    );
-  }
+  const countryProvinceSelector = new CountryProvinceSelector(window.theme.allCountryOptionTags);
+
+  countryProvinceSelector.build(
+    $('#AddressCountryNew')[0],
+    $('#AddressProvinceNew')[0],
+    {
+      hideClass: 'hide',
+      hideElement: $('#AddressProvinceContainerNew')[0],
+    },
+  );
 
   // Initialize each edit form's country/province selector
   $('.address-country-option').each(function() {
     const formId = $(this).data('form-id');
-    const countrySelector = `AddressCountry_${formId}`;
-    const provinceSelector = `AddressProvince_${formId}`;
-    const containerSelector = `AddressProvinceContainer_${formId}`;
+    const countrySelector = $(`#AddressCountry_${formId}`)[0];
+    const provinceSelector = $(`#AddressProvince_${formId}`)[0];
+    const containerSelector = $(`#AddressProvinceContainer_${formId}`)[0];
 
-    new Shopify.CountryProvinceSelector(countrySelector, provinceSelector, {
-      hideElement: containerSelector,
-    });
+    countryProvinceSelector.build(
+      countrySelector,
+      provinceSelector,
+      {
+        hideClass: 'hide',
+        hideElement: containerSelector,
```

Comment:
The fact that we need to specify and element and a class makes me feel like this approach is limited and is making the assumption that the user only wants to apply a class directly to a DOM element. 

Would it be more flexible for the developer to provide a callback function that can be executed on country change? Or only when provinces.length === 0? That way they can perform whatever operation they want?

Reply:
After some discussion with @t-kelly, we decided to split the functionality and UI concerns. This means that we will provide on `change` callbacks for developers to do whatever they need to do with the given parameters.
- `theme-address` package should only concern with the building and populating the `<select>` pairs
- on `change` callbacks gives the developers the freedom to do anything they need on the user interface side, like hiding the province selector or its wrapper